app/src/UI/Features/LegacyMap/helpers/functional/layers-hook.ts

```diff
@@ -0,0 +1,195 @@
+  const features = useSelector((state) => state.Configuration.current.features);
+
+  /* evaluate which layers are currently available to select */
+  useEffect(() => {
+    const newFilteredLayerDefinitions: InvasivesMapLayerDefinitionWithState[] = [];
+
+    // evaluate each potential map definition and remove those not eligible at this moment
+    for (const l of [...MAP_DEFINITIONS, ...(offlineDefinitions ?? [])] as InvasivesMapLayerDefinition[]) {
+      let pass = true;
+
+      if (!l.predicates.directlySelectable) {
+        pass = false;
+      }
+
+      if (l.predicates.mobileOnly && !MOBILE) {
+        pass = false;
+      }
+
+      if (l.predicates.webOnly && MOBILE) {
+        pass = false;
+      }
+
+      if (l.predicates.requiresPlatform !== undefined) {
+        if (l.predicates.requiresPlatform !== platform) {
+          pass = false;
+        }
+      }
+
+      if (l.predicates.requiresFeature !== undefined) {
+        if (!features[l.predicates.requiresFeature].enabled) {
+          pass = false;
+        }
+      }
+
+      if (l.predicates.requiresAuthentication && !loggedInOrWorkingOffline) {
+        pass = false;
+      }
+
+      if (l.predicates.requiresAnonymous && loggedInOrWorkingOffline) {
+        pass = false;
+      }
+
+      if (l.predicates.requiresNetwork && !connected) {
+        pass = false;
+      }
+
+      if (!l.predicates.requiresNetwork && l.predicates.mobileOnly && connected) {
+        pass = false;
+      }
+
+      if (pass) {
+        newFilteredLayerDefinitions.push({
+          active: false,
+          ...l
+        });
+      }
+    }
+
+    setAvailableLayerDefinitions(newFilteredLayerDefinitions);
+  }, [features, connected, offlineDefinitions, MOBILE, platform, loggedInOrWorkingOffline]);
+
+  /* ensure that at least one basemap layer is always designated as active */
+  useEffect(() => {
+    if (availableLayerDefinitions.length == 0) return;
+    if (availableLayerDefinitions.filter((l) => l.mode === 'basemap').length == 0) return;
+    if (availableLayerDefinitions.filter((l) => l.mode === 'basemap' && l.active).length > 0) return;
+
+    setAvailableLayerDefinitions(
+      produce((draft) => {
+        const found = draft.find((l) => l.mode === 'basemap');
+        if (found) {
+          found.active = true;
+        }
+      })
+    );
+  }, [availableLayerDefinitions]);
+
+  /* set the state of the public vector layer correctly on auth state change */
+  useEffect(() => {
+    if (availableLayerDefinitions.length == 0) return;
+    if (availableLayerDefinitions.filter((l) => l.name === 'Public-Vector').length == 0) return;
+    if (
+      availableLayerDefinitions.filter((l) => l.name === 'Public-Vector' && l.active == !loggedInOrWorkingOffline)
+        .length > 0
+    )
+      return;
+
+    const updatedLayerDefinitions: InvasivesMapLayerDefinitionWithState[] = [];
+
+    for (const l of availableLayerDefinitions) {
+      const updated = l;
```

Comment:
```suggestion
      const updated = { ...l };
```
This seems to correct the issue of immutability. good 'ol memory references